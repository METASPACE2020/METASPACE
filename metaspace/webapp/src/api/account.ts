const fetchPostJson = async (url: string, body: object) =>
  await fetch(url, {
    method: 'POST',
    headers: {'Content-Type': 'application/json'},
    body: JSON.stringify(body),
  });

export const signInByEmail = async (email: string, password: string): Promise<boolean> => {
  const response = await fetchPostJson('/auth/signin', { email, password });
  return response.status >= 200 && response.status < 300;
};

export const createAccountByEmail = async (email: string, password: string, name: string) => {
  const response = await fetchPostJson('/auth/createaccount', { email, password, name });
  if (response.status !== 200) {
    throw new Error(`Unexpected response from server: ${response.status} ${response.statusText}`)
  }
};

export const sendPasswordResetToken = async (email: string) => {
  const response = await fetchPostJson('/auth/sendpasswordresettoken', { email });
  if (response.status !== 200) {
    throw new Error(`Unexpected response from server: ${response.status} ${response.statusText}`)
  }
};

export const validatePasswordResetToken = async (token: string, email: string): Promise<boolean> => {
  const response = await fetchPostJson('/auth/validatepasswordresettoken', { token, email });
  return response.status === 200;
};


export const resetPassword = async (token: string, email: string, password: string) => {
  const response = await fetchPostJson('/auth/sendpasswordreset', { token, email, password });
  if (response.status !== 200) {
    throw new Error(`Unexpected response from server: ${response.status} ${response.statusText}`)
  }
};
