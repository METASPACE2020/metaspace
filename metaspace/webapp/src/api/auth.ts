const fetchPostJson = async (url: string, body: object) =>
  fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    credentials: 'include',
    body: JSON.stringify(body),
  })

export const signOut = async () => {
  const response = await fetch('/api_auth/signout', { method: 'POST', credentials: 'include' })
  if (response.status !== 200) {
    throw new Error(`Unexpected response from server: ${response.status} ${response.statusText}`)
  }
}

export const signInByEmail = async (email: string, password: string): Promise<boolean> => {
  const response = await fetchPostJson('/api_auth/signin', { email, password })
  return response.status >= 200 && response.status < 300
}

export const verifyRecaptcha = async (recaptchaToken: string): Promise<boolean> => {
  const response = await fetchPostJson('/api_auth/verify_captcha', { recaptchaToken })
  return response.status >= 200 && response.status < 300
}

export const createAccountByEmail = async (email: string, password: string, name: string, recaptchaToken?: string) => {
  const response = await fetchPostJson('/api_auth/createaccount', {
    email,
    password,
    name,
    recaptchaToken,
  })
  if (response.status !== 200) {
    let error = `Unexpected response from server: ${response.status} ${response.statusText}`
    try {
      const data = await response.json()
      error = JSON.parse(data.message).message
    } catch (error) {
      // ignore
    }

    throw new Error(error)
  }
}

export const sendPasswordResetToken = async (email: string) => {
  const response = await fetchPostJson('/api_auth/sendpasswordresettoken', { email })
  if (response.status !== 200) {
    throw new Error(`Unexpected response from server: ${response.status} ${response.statusText}`)
  }
}

export const validatePasswordResetToken = async (token: string, email: string): Promise<boolean> => {
  const response = await fetchPostJson('/api_auth/validatepasswordresettoken', { token, email })
  return response.status === 200
}

export const resetPassword = async (token: string, email: string, password: string) => {
  const response = await fetchPostJson('/api_auth/resetpassword', { token, email, password })
  if (response.status !== 200) {
    throw new Error(`Unexpected response from server: ${response.status} ${response.statusText}`)
  }
}
