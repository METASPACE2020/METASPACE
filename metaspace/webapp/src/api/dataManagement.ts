import gql from 'graphql-tag';

export interface CurrentUserResult {
  id: string;
  name: string;
  email: string | null;
  groups: {
    role: string;
    group: {
      id: string,
      name: string
    };
  }[]
  primaryGroup: {
    group: {
      id: string,
      name: string
    }
  } | null;
}

export const currentUserQuery =
gql`query {
  currentUser {
    id
    name
    role
    email
    groups {
      role
      group {
        id
        name
      }
    }
    primaryGroup {
      group {
        id
        name
      }
    }
  }
}
`;

export interface GroupListItem {
  id: string;
  name: string;
}

export const oneGroupQuery =
  gql`query($groupId: ID!) {
  group(groupId:$groupId) {
    id
    name
  }
}`;

export const allGroupsQuery =
gql`query($query: String) {
  allGroups(query:$query) {
    id
    name
  }
}`;

export const requestAccessToGroupMutation =
gql`mutation($groupId: ID!) {
  requestAccessToGroup(groupId: $groupId){
    role
    }
  }`;

