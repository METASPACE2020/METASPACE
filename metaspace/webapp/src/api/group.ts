import gql from 'graphql-tag';

export type UserGroupRole = 'INVITED' | 'PENDING' | 'MEMBER' | 'PRINCIPAL_INVESTIGATOR';



export const createGroupMutation =
  gql`mutation createGroup($groupDetails: CreateGroupInput!) {
    createGroup(groupDetails: $groupDetails) {
      id
      name
      shortName
    }
  }`;


export const updateGroupMutation =
  gql`mutation updateGroup($id: ID!, $groupDetails: UpdateGroupInput!) {
    updateGroup(id: $id, groupDetails: $groupDetails) {
      id
      name
      shortName
    }
  }`;

export const deleteGroupMutation =
  gql`mutation deleteGroup($id: ID!) {
    deleteGroup(id: $id)
  }`;

export const removeUserFromGroupMutation =
  gql`mutation removeUserFromGroup($groupId: ID!, $userId: ID!) {
    removeUserFromGroup(groupId: $groupId, userId: $userId)
  }`;

export const requestAccessToGroupMutation =
  gql`mutation requestAccessToGroup($groupId: ID!, $bringDatasets: [ID!]!) { 
    requestAccessToGroup(groupId: $groupId, bringDatasets: $bringDatasets) {
      role # GraphQL demands a return value be specified
    }
  }`;

export const acceptRequestToJoinGroupMutation =
  gql`mutation acceptRequestToJoinGroup($groupId: ID!, $userId: ID!) {
    acceptRequestToJoinGroup(groupId: $groupId, userId: $userId) {
      role
    }
  }`;

export const inviteUserToGroupMutation =
  gql`mutation inviteUserToGroup($groupId: ID!, $email: String!) {
    inviteUserToGroup(groupId: $groupId, email: $email) {
      role
    }
  }`;

export const acceptGroupInvitationMutation =
  gql`mutation acceptGroupInvitation($groupId: ID!, $bringDatasets: [ID!]!) { 
    acceptGroupInvitation(groupId: $groupId, bringDatasets: $bringDatasets) {
      role
    }
  }`;

export const leaveGroupMutation =
  gql`mutation leaveGroup($groupId: ID!) { 
    leaveGroup(groupId: $groupId)
  }`;


export const editGroupQuery =
  gql`query EditGroupProfileQuery($groupId: ID!) {
    group(id: $groupId) {
      id
      name
      shortName
      members {
        role
        numDatasets
        user {
          id
          name
          email
        }
      }
    }
  }`;

export interface EditGroupQuery {
  id: string;
  name: string;
  shortName: string;
  members: EditGroupQueryMember[] | null;
}
export interface EditGroupQueryMember {
  role: UserGroupRole,
  numDatasets: number,
  user: EditGroupQueryUser
}
export interface EditGroupQueryUser {
  id: string;
  name: string;
  email: string | null;
}
