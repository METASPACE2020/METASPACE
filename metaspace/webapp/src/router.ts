/* eslint-disable vue/max-len */
import Vue, { AsyncComponent } from 'vue'
import VueRouter, { RawLocation } from 'vue-router'
import AboutPage from './modules/App/AboutPage.vue'
import DatasetsPage from './modules/Datasets/DatasetsPage.vue'
import { DialogPage, ResetPasswordPage } from './modules/Account'
import { redirectAfterSignIn } from './modules/Account/signInReturnUrl'
import { updateDBParam } from './modules/Filters/url'
import NotFoundPage from './modules/App/NotFoundPage.vue'

Vue.use(VueRouter)

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Annotations/AnnotationsPage.vue'),
  DatasetSummary: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetSummary" */ './modules/Datasets/summary/DatasetSummary.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ './modules/MetadataEditor/MetadataEditPage.vue'),
  ImageAlignmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "ImageAlignmentPage" */ './modules/ImageAlignment/ImageAlignmentPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ './modules/MetadataEditor/UploadPage.vue'),
  DesignPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DesignPage" */ './design/DesignPage.vue'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/HelpPage.vue'),
  EditUserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/UserProfile/EditUserPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/GroupProfile/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ProjectsListPage.vue'),
  SystemHealthPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/SystemHealthPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/GroupsListPage.vue'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PrivacyPage.vue'),
  TermsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/TermsPage.vue'),
  PublicationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PublicationsPage.vue'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/GroupProfile/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Project/ViewProjectPage.vue'),
}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, AsyncComponent>

const convertLegacyUrls = () => {
  const { pathname, hash, search } = window.location
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined as any, hash.slice(1))
  }
  if (pathname === '/annotations') {
    const updatedQueryString = updateDBParam(search.slice(1))
    if (updatedQueryString !== null) {
      history.replaceState(undefined, undefined as any, `${pathname}?${updatedQueryString}`)
    }
  }
}
convertLegacyUrls()

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/(about)?', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
    { path: '/annotations', component: asyncPages.AnnotationsPage },
    {
      path: '/datasets',
      component: DatasetsPage,
      children: [
        { path: '', component: asyncPages.DatasetTable },
        { path: 'summary', component: asyncPages.DatasetSummary },
      ],
    },
    { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
    { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.ImageAlignmentPage },
    { path: '/upload', component: asyncPages.UploadPage },
    { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },
    { path: '/user/me', component: asyncPages.EditUserPage },

    { path: '/admin/health', component: asyncPages.SystemHealthPage },
    { path: '/admin/groups', component: asyncPages.GroupsListPage },

    { path: '/account/sign-in', component: DialogPage, props: { dialog: 'signIn' } },
    { path: '/account/sign-in-success', redirect: redirectAfterSignIn },
    { path: '/account/create-account', component: DialogPage, props: { dialog: 'createAccount' } },
    { path: '/account/forgot-password', component: DialogPage, props: { dialog: 'forgotPassword' } },
    { path: '/account/reset-password', component: ResetPasswordPage },

    { path: '/group/create', component: asyncPages.CreateGroupPage },
    { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },

    { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
    { // Legacy URL sent in "request access" emails up until Feb 2019
      path: '/project/:projectIdOrSlug/manage',
      redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } } as RawLocation,
    },
    { path: '/projects', component: asyncPages.ProjectsListPage },

    { path: '/terms', component: asyncPages.TermsPage, meta: { footer: true } },
    { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },
    { path: '/publications', component: asyncPages.PublicationsPage, meta: { footer: true } },

    { path: '/design', component: asyncPages.DesignPage, meta: { footer: true, flex: true } },

    { path: '*', component: NotFoundPage, meta: { footer: true, flex: true } },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) {
      return { selector: to.hash, offset: { x: 0, y: 64 } } // offset header height
    }
    if (savedPosition) {
      return savedPosition
    }
    if (to.path !== from.path) {
      return { x: 0, y: 0 }
    }
  },
})

const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined, true)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
