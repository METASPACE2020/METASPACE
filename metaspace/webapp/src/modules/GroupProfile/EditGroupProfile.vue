<template>
  <div>
    <div class="page">
      <div class="page-content">
        <div class="header-row">
          <h1>Group Details</h1>
          <div class="flex-spacer" />

          <div class="header-row-buttons">
            <el-button v-if="canEdit && group"
                       type="primary"
                       @click="handleSave"
                       :loading="isSaving">
              Save
            </el-button>
          </div>
        </div>
        <edit-group-form :model="model" :disabled="isSaving || !canEdit" />
        <members-list
          :loading="groupLoading !== 0"
          :members="group && group.members || []"
          type="group"
          :filter="datasetsListFilter"
          :canEdit="canEdit"
          @removeUser="handleRemoveUser"
          @cancelInvite="handleRemoveUser"
          @acceptUser="handleAcceptUser"
          @rejectUser="handleRejectUser"
          @addMember="handleAddMember"
        />
        <div style="margin-bottom: 2em">
          <h2>Datasets</h2>
          <p>
            <router-link :to="datasetsListLink">See all datasets</router-link>
          </p>
        </div>
        <div v-if="canDelete && group">
          <h2>Delete group</h2>
          <p>
            Please ensure all datasets have been removed before deleting a group.
          </p>
          <div style="text-align: right; margin: 1em 0;">
            <el-button
              type="danger"
              @click="handleDeleteGroup"
              :loading="isDeletingGroup">
              Delete group
            </el-button>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
<script lang="ts">
  import Vue from 'vue';
  import { Component, Watch } from 'vue-property-decorator';
  import {
    acceptRequestToJoinGroupMutation,
    deleteGroupMutation,
    editGroupQuery,
    EditGroupQuery,
    EditGroupQueryMember,
    inviteUserToGroupMutation,
    removeUserFromGroupMutation,
    UpdateGroupMutation,
    updateGroupMutation,
  } from '../../api/group';
  import gql from 'graphql-tag';
  import EditGroupForm from './EditGroupForm.vue';
  import MembersList from '../../components/MembersList.vue';
  import { UserRole } from '../../api/user';
  import { encodeParams } from '../Filters';
  import ConfirmAsync from '../../components/ConfirmAsync';
  import reportError from '../../lib/reportError';
  import emailRegex from '../../lib/emailRegex';

  interface CurrentUserQuery {
    id: string;
    role: UserRole;
  }

  @Component({
    components: {
      EditGroupForm,
      MembersList,
    },
    apollo: {
      currentUser: gql`query {
        currentUser {
          id
          role
        }
      }`,
      group: {
        query: editGroupQuery,
        loadingKey: 'membersLoading',
        variables(this: EditGroupProfile) { return { groupId: this.groupId } },
      },
    }
  })
  export default class EditGroupProfile extends Vue {
    groupLoading = 0;
    isDeletingGroup = false;
    isSaving = false;
    model = {
      name: '',
      shortName: '',
    };

    currentUser: CurrentUserQuery | null = null;
    group: EditGroupQuery | null = null;

    get canDelete(): boolean {
      return this.currentUser && this.currentUser.role === 'admin' || false;
    }
    get canEdit(): boolean {
      return (this.currentUser && this.currentUser.role === 'admin')
        || (this.group && this.group.currentUserRole === 'PRINCIPAL_INVESTIGATOR')
        || false;
    }
    get groupId(): string {
      return this.$route.params.groupId;
    }
    get groupName() {
      return this.group ? this.group.name : '';
    }
    get datasetsListFilter() {
      return {
        group: this.groupId,
      };
    }

    @Watch('group')
    setModel() {
      this.model.name = this.group && this.group.name || '';
      this.model.shortName = this.group && this.group.shortName || '';
    }

    get datasetsListLink() {
      return { path: '/datasets', query: encodeParams(this.datasetsListFilter) }
    }

    @ConfirmAsync(function (this: EditGroupProfile) {
      return {
        message: `Are you sure you want to delete ${this.groupName}?`,
        confirmButtonText: 'Delete group',
        confirmButtonLoadingText: 'Deleting...'
      }
    })
    async handleDeleteGroup() {
      this.isDeletingGroup = true;
      try {
        const groupName = this.groupName;
        await this.$apollo.mutate({
          mutation: deleteGroupMutation,
          variables: { groupId: this.groupId },
        });
        this.$message({ message: `${groupName} has been deleted`, type: 'success' });
        this.$router.push('/');
      } catch(err) {
        reportError(err);
      } finally {
        this.isDeletingGroup = false;
      }
    }

    async handleSave() {
      this.isSaving = true;
      try {
        const {name, shortName} = this.model;
        await this.$apollo.mutate<UpdateGroupMutation>({
          mutation: updateGroupMutation,
          variables: { groupId: this.groupId, groupDetails: { name, shortName } },
        });
        this.$message({ message: `${name} has been saved`, type: 'success' });
      } catch(err) {
        reportError(err);
      } finally {
        this.isSaving = false;
      }
    }

    @ConfirmAsync(function (this: EditGroupProfile, member: EditGroupQueryMember) {
      return {
        message: `Are you sure you want to remove ${member.user.name} from ${this.groupName}?`,
        confirmButtonText: 'Remove user',
        confirmButtonLoadingText: 'Removing...'
      }
    })
    async handleRemoveUser(member: EditGroupQueryMember) {
      await this.$apollo.mutate({
        mutation: removeUserFromGroupMutation,
        variables: { groupId: this.groupId, userId: member.user.id },
      });
      await this.$apollo.queries.group.refetch();
    }

    @ConfirmAsync(function (this: EditGroupProfile, member: EditGroupQueryMember) {
      return {
        message: `This will allow ${member.user.name} to access all private datasets that are in ${this.groupName}. Are you sure you want to accept them into the group?`,
        confirmButtonText: 'Accept request',
        confirmButtonLoadingText: 'Accepting...'
      }
    })
    async handleAcceptUser(member: EditGroupQueryMember) {
      await this.$apollo.mutate({
        mutation: acceptRequestToJoinGroupMutation,
        variables: { groupId: this.groupId, userId: member.user.id },
      });
      await this.$apollo.queries.group.refetch();
    }

    @ConfirmAsync(function (this: EditGroupProfile, member: EditGroupQueryMember) {
      return {
        message: `Are you sure you want to decline ${member.user.name}'s request for access to ${this.groupName}?`,
        confirmButtonText: 'Decline request',
        confirmButtonLoadingText: 'Declining...'
      }
    })
    async handleRejectUser(member: EditGroupQueryMember) {
      await this.$apollo.mutate({
        mutation: removeUserFromGroupMutation,
        variables: { groupId: this.groupId, userId: member.user.id },
      });
      await this.$apollo.queries.group.refetch();
    }

    @ConfirmAsync(function (this: EditGroupProfile) {
      return {
        title: 'Add member',
        message: `An email will be sent inviting them to join the group. If they accept the invitation, they will be able to access the private datasets of ${this.groupName}.`,
        showInput: true,
        inputPlaceholder: 'Email address',
        inputPattern: emailRegex,
        inputErrorMessage: 'Please enter a valid email address',
        confirmButtonText: 'Invite to group',
        confirmButtonLoadingText: 'Sending invitation...'
      }
    })
    async handleAddMember(email: string) {
      await this.$apollo.mutate({
        mutation: inviteUserToGroupMutation,
        variables: { groupId: this.groupId, email },
      });
      await this.$apollo.queries.group.refetch();
    }
  }

</script>
<style scoped lang="scss">
  .page {
    display: flex;
    justify-content: center;
    min-height: 80vh; // Ensure there's space for the loading spinner before is visible
  }

  .page-content {
    width: 950px;
  }

  .header-row {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
  }

  .header-row-buttons {
    display: flex;
    margin-right: 3px;
  }

  .flex-spacer {
    flex-grow: 1;
  }

</style>
