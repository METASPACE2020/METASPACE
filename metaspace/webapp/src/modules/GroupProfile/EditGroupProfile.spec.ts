import { mount } from '@vue/test-utils';
import Vue from 'vue';
import EditGroupProfile from './EditGroupProfile.vue';
import router from '../../router';
import { EditGroupQuery } from '../../api/group';


describe('EditGroupProfile', () => {

  const currentUser = {id:'1', role:'user'};
  const mockGroup: EditGroupQuery = {
    id: '2',
    name: 'European Molecular Biology Laboratory',
    shortName: 'EMBL',
    currentUserRole: 'PRINCIPAL_INVESTIGATOR',
    members: [
      {
        role: 'PRINCIPAL_INVESTIGATOR',
        numDatasets: 123,
        user: {
          id: '3',
          name: 'me',
          email: 'my-email@example.com'
        }
      },
      {
        role: 'PENDING',
        numDatasets: 0,
        user: {
          id: '4',
          name: 'Person who asked to join',
          email: 'access@requestor.com'
        }
      },
      {
        role: 'INVITED',
        numDatasets: 0,
        user: {
          id: '5',
          name: 'Invitee',
          email: 'awaiting@response.com'
        }
      },
      {
        role: 'MEMBER',
        numDatasets: 1,
        user: {
          id: '6',
          name: 'Group member',
          email: 'person@embl.de'
        }
      }
    ]
  };

  router.replace({ name: 'edit-group', params: { groupId: mockGroup.id } });

  it('should match snapshot', async () => {
    const wrapper = mount(EditGroupProfile, { router, sync: false });
    wrapper.setData({
      currentUser,
      group: mockGroup
    });

    await Vue.nextTick();

    expect(wrapper).toMatchSnapshot();
  });
});
