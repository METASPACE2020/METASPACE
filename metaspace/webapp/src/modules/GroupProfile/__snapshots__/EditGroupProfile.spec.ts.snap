// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditGroupProfile should match snapshot 1`] = `
<div>
  <div class="page">
    <div class="page-content">
      <div class="header-row">
        <h1>Group Details</h1>
        <div class="flex-spacer"></div>
        <div class="header-row-buttons">
          <!---->
          <button type="button" class="el-button el-button--primary">
            <!---->
            <!----><span>
            Save
          </span></button>
        </div>
      </div>
      <form class="el-form el-form--label-top">
        <div>
          <div class="el-form-item name is-required">
            <label for="name" class="el-form-item__label">Full name</label>
            <div class="el-form-item__content">
              <div class="el-input">
                <!---->
                <input type="text" autocomplete="off" maxlength="50" class="el-input__inner">
                <!---->
                <!---->
                <!---->
              </div>
              <!---->
            </div>
          </div>
          <div class="el-form-item shortName is-required">
            <label for="shortName" class="el-form-item__label"><span>
        Short name
        <mock-el-popover trigger="hover" placement="right"><div slot-key="default"> <p>The short name will be shown whenever space is limited.</p></div><div slot-key="reference"><i class="el-icon-question metadata-help-icon"></i></div></mock-el-popover></span></label>
            <div
              class="el-form-item__content">
              <div class="el-input">
                <!---->
                <input type="text" autocomplete="off" maxlength="20" class="el-input__inner">
                <!---->
                <!---->
                <!---->
              </div>
              <!---->
          </div>
        </div>
    </div>
    <!---->
    </form>
    <div>
      <h1>Members</h1>
      <div class="el-table el-table--fit el-table--scrollable-x el-table--enable-row-hover el-table--enable-row-transition" element-loading-text="Loading results from the server...">
        <div class="hidden-columns">
          <div></div>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
        </div>
        <div class="el-table__header-wrapper">
          <table cellspacing="0" cellpadding="0" border="0" class="el-table__header" style="width: 850px;">
            <colgroup>
              <col name="el-table_1_column_1" width="200">
              <col name="el-table_1_column_2" width="200">
              <col name="el-table_1_column_3" width="160">
              <col name="el-table_1_column_4" width="90">
              <col name="el-table_1_column_5" width="200">
            </colgroup>
            <thead class="">
              <tr class="">
                <th colspan="1" rowspan="1" class="el-table_1_column_1     is-leaf">
                  <div class="cell">Name</div>
                </th>
                <th colspan="1" rowspan="1" class="el-table_1_column_2     is-leaf">
                  <div class="cell">Email</div>
                </th>
                <th colspan="1" rowspan="1" class="el-table_1_column_3     is-leaf">
                  <div class="cell">Role</div>
                </th>
                <th colspan="1" rowspan="1" class="el-table_1_column_4  is-center   is-leaf">
                  <div class="cell">Datasets</div>
                </th>
                <th colspan="1" rowspan="1" class="el-table_1_column_5     is-leaf">
                  <div class="cell"></div>
                </th>
              </tr>
            </thead>
          </table>
        </div>
        <div class="el-table__body-wrapper is-scrolling-left">
          <table cellspacing="0" cellpadding="0" border="0" class="el-table__body" style="width: 850px;">
            <colgroup>
              <col name="el-table_1_column_1" width="200">
              <col name="el-table_1_column_2" width="200">
              <col name="el-table_1_column_3" width="160">
              <col name="el-table_1_column_4" width="90">
              <col name="el-table_1_column_5" width="200">
            </colgroup>
            <tbody>
              <tr class="el-table__row">
                <td rowspan="1" colspan="1" class="el-table_1_column_1  ">
                  <div class="cell">
                    me
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_2  ">
                  <div class="cell">
                    my-email@example.com
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_3  ">
                  <div class="cell">
                    Principal investigator
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_4 is-center ">
                  <div class="cell">
                    <a href="#/datasets?grp=%7B%22id%22%3A%222%22,%22name%22%3A%22European%20Molecular%20Biology%20Laboratory%22%7D&amp;subm=%7B%22id%22%3A%223%22,%22name%22%3A%22me%22%7D" class="">123</a>
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_5  ">
                  <div class="cell">
                    <!---->
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </td>
              </tr>
              <tr class="el-table__row">
                <td rowspan="1" colspan="1" class="el-table_1_column_1  ">
                  <div class="cell">
                    Person who asked to join
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_2  ">
                  <div class="cell">
                    access@requestor.com
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_3  ">
                  <div class="cell">
                    Requesting access
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_4 is-center ">
                  <div class="cell">
                    <a href="#/datasets?grp=%7B%22id%22%3A%222%22,%22name%22%3A%22European%20Molecular%20Biology%20Laboratory%22%7D&amp;subm=%7B%22id%22%3A%224%22,%22name%22%3A%22Person%20who%20asked%20to%20join%22%7D" class="">0</a>
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_5  ">
                  <div class="cell">
                    <!---->
                    <!---->
                    <button type="button" class="el-button grid-button el-button--success el-button--mini">
                      <!---->
                      <!----><span>
          Accept
        </span></button>
                    <button type="button" class="el-button grid-button el-button--default el-button--mini">
                      <!---->
                      <!----><span>
          Decline
        </span></button>
                  </div>
                </td>
              </tr>
              <tr class="el-table__row">
                <td rowspan="1" colspan="1" class="el-table_1_column_1  ">
                  <div class="cell">
                    Invitee
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_2  ">
                  <div class="cell">
                    awaiting@response.com
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_3  ">
                  <div class="cell">
                    Invited
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_4 is-center ">
                  <div class="cell">
                    <a href="#/datasets?grp=%7B%22id%22%3A%222%22,%22name%22%3A%22European%20Molecular%20Biology%20Laboratory%22%7D&amp;subm=%7B%22id%22%3A%225%22,%22name%22%3A%22Invitee%22%7D" class="">0</a>
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_5  ">
                  <div class="cell">
                    <!---->
                    <button type="button" class="el-button grid-button el-button--default el-button--mini">
                      <!---->
                      <!----><span>
          Cancel
        </span></button>
                    <!---->
                    <!---->
                  </div>
                </td>
              </tr>
              <tr class="el-table__row">
                <td rowspan="1" colspan="1" class="el-table_1_column_1  ">
                  <div class="cell">
                    Group member
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_2  ">
                  <div class="cell">
                    person@embl.de
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_3  ">
                  <div class="cell">
                    Member
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_4 is-center ">
                  <div class="cell">
                    <a href="#/datasets?grp=%7B%22id%22%3A%222%22,%22name%22%3A%22European%20Molecular%20Biology%20Laboratory%22%7D&amp;subm=%7B%22id%22%3A%226%22,%22name%22%3A%22Group%20member%22%7D" class="">1</a>
                  </div>
                </td>
                <td rowspan="1" colspan="1" class="el-table_1_column_5  ">
                  <div class="cell">
                    <button type="button" class="el-button grid-button el-button--default el-button--mini">
                      <!---->
                      <!----><span>
          Remove
        </span></button>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </td>
              </tr>
              <!---->
            </tbody>
          </table>
          <!---->
          <!---->
        </div>
        <!---->
        <!---->
        <!---->
        <!---->
        <div class="el-table__column-resize-proxy" style="display: none;"></div>
      </div>
      <div class="pagination-row">
        <!---->
        <div class="flex-spacer"></div>
        <button type="button" class="el-button el-button--default">
          <!---->
          <!----><span>Add member</span></button>
      </div>
    </div>
    <div>
      <h1>Datasets</h1>
      <a href="#/datasets?grp=%7B%22name%22%3A%22European%20Molecular%20Biology%20Laboratory%22%7D" class="">See all datasets</a>
    </div>
  </div>
</div>
</div>
`;
